SingularityUI/app/components/common/Table.cjsx

```diff
@@ -0,0 +1,47 @@
+Utils = require '../../utils'
+
+Table = React.createClass
+
+    renderTableHeader: ->
+        header = []
+        key = 0
+        for columnName in @props.columnNames
+            header.push(<th key={key}>{columnName}</th>)
+            key++
+        return header
+
+    renderTableRow: (elements) ->
+        data = []
+        key = 0
+        for element in elements
+            data.push(<td key={key}>{element}</td>)
+            key++
+        return data
+
+    # Note: The data in @props.tableRows should be the JSX you would like to render
```

Comment:
If you prefer, you can instead use the pattern where you pass the name of a specific row component as a prop into this `Table` component, as well as a prop that contains raw data (instead of straight `jsx`). You can call the row component by name:

```
TableRow = @props.rowComponent # e.g. @props.rowComponent = "MyTableRowComponent"

...

<TableRow # is replaced with "MyTableRowComponent"
  prop={@props.prop}
  ...
/>
```